src/Adapter/Order/CommandHandler/CancelOrderProductHandler.php

```diff
@@ -114,29 +110,17 @@ public function handle(CancelOrderProductCommand $command)
         if (!empty($orderDetails['productsOrderDetails'])) {
             foreach ($orderDetails['productsOrderDetails'] as $orderDetail) {
                 $qty_cancel_product = $orderDetails['productCancelQuantity'][$orderDetail->id_order_detail];
-                $this->reinjectQuantity($orderDetail, $qty_cancel_product);
-
-                $this->orderProductRemover->deleteProductFromOrder($order, $orderDetail, $qty_cancel_product);
-
-                // Update weight SUM
-                $order_carrier = new OrderCarrier((int) $order->getIdOrderCarrier());
-                if (Validate::isLoadedObject($order_carrier)) {
-                    $order_carrier->weight = (float) $order->getTotalWeight();
-                    if ($order_carrier->update()) {
-                        $order->weight = sprintf('%.3f ' . Configuration::get('PS_WEIGHT_UNIT'), $order_carrier->weight);
-                    }
-                }
-
-                if (Configuration::get('PS_ADVANCED_STOCK_MANAGEMENT') && StockAvailable::dependsOnStock($orderDetail->product_id)) {
-                    StockAvailable::synchronize($orderDetail->product_id);
-                }
+                $newQuantity = (int) $orderDetail->product_quantity - (int) $qty_cancel_product;
+                $this->orderProductQuantityUpdater->update($order, $orderDetail, $newQuantity, null);
```

Comment:
You use null as the last parameter, but the OrderDetail can have an invoice associated

Reply:
If the order has an invoice then the user should not be able to perform a "cancel product", this case should never happen.